docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+echo -n "observability-javalin-app:ZXUtY2VudHJhbC0xLmF3cy5jbG91ZC5lcy5pbyQ4NDU5M2I1YmQzYTY0N2NhYjA2MWQ3NTJhZWFhNWEzYyQzYmQwMWE2OTQ2MmQ0N2ExYjdhYTkwMzI0YjJiOTMyYQ==" | ./metricbeat keystore add CLOUD_ID --stdin
+----
++
+. Configure {metricbeat} to read our Prometheus metrics. Start with a
+basic `metricbeat.yaml`.
++
+[source,yaml]
+----
+metricbeat.config.modules:
+  path: ${path.config}/modules.d/\*.yml
+  reload.enabled: false
+
+name: javalin-metrics-shipper
+
+cloud.id: ${CLOUD_ID}
+output.elasticsearch:
+  api_key: ${ES_API_KEY}
+
+processors:
+  - add_host_metadata: ~
+  - add_cloud_metadata: ~
+  - add_docker_metadata: ~
+  - add_kubernetes_metadata: ~
+----
++
+As {metricbeat} supports dozens of modules, which in turn are different
+ways of ingesting metrics (the same applies to {filebeat} with different
+types of log files and formats), the Prometheus module needs to be enabled.
++
+[source,bash]
+----
+./metricbeat modules enable prometheus
+----
++
+Add the Prometheus endpoint to poll.
+`./modules.d/prometheus.yml`:
++
+[source,yaml]
+----
+- module: prometheus
+  period: 10s
+  hosts: ["localhost:7000"]
+  metrics_path: /metrics
+  username: "metrics"
+  password: "secret"
+  use_types: true
+  rate_counters: true
+----
++
+To improve security, add the username and the
+password to the keystore and refer it here as well.
++
+. Start {metricbeat} and verify that the Prometheus events are flowing into
+{es}.
++
+[source,console]
+----
+GET metricbeat-7.9.0/_search?filter_path=**.prometheus,hits.total
+{
+  "query": {
+    "term": {
+      "event.module": "prometheus"
+    }
+  }
+}
+----
+
+[discrete]
+== Step 5: View metrics in {kib}
+
+As this is custom data from our Javalin app, there is no pre-defined
+dashboard for displaying this data.
+
+Let’s check for the number of logging messages per log level.
+
+[source,console]
+----
+GET metricbeat-7.9.0/_search
+{
+  "query": {
+    "exists": {
+      "field": "prometheus.log4j2_events_total.counter"
+    }
+  }
+}
+----
+
+Visualize the number of log messages over time, split by the
+log level. Since the Elastic Stack 7.7, there is a new way of creating a
+visualization called `Lens`.
+
+. Log into {kib} and select *Visualize* -> *Create Visualization*.
++
+. Create a line chart and select `metricbeat-*` as the source. 
++
+The basic idea is to have a
+https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-max-aggregation.html[max
+aggregation] on the y-axis on the `prometheus.log4j2_events_total.rate`
+field, whereas the x-axis, is split by date using a
+https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-datehistogram-aggregation.html[date_histogram
+aggregation] on the `@timestamp` field.
++
+There is one more split within
+each date histogram bucket, split by log level, using a
+https://www.elastic.co/guide/en/elasticsearch/reference/7.8/search-aggregations-bucket-terms-aggregation.html[terms
```

Comment:
```suggestion
{ref}/search-aggregations-bucket-terms-aggregation.html[terms
```

Reply:
I haven't tested the link that this resolves to, but I'm assuming you aren't intentionally pointing to 7.8 here. It's best to let the shared attributes file resolve the branch.